### PR TITLE
games-emulation/dosbox: fix compiling with clang

### PR DESCRIPTION
**Pull request by austin987**
Gentoo-Bug: https://bugs.gentoo.org/449060

Package-Manager: portage-2.2.26

### CODE REVIEW

File: games-emulation/dosbox/dosbox-0.74-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Comment (monsieurp) [inline games-emulation/dosbox/dosbox-0.74-r1.ebuild]: Why not 6?

File: games-emulation/dosbox/dosbox-0.74-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	opengl? ( virtual/glu virtual/opengl )
+	debug? ( sys-libs/ncurses:0 )
+	media-libs/libpng:0
+	media-libs/libsdl[joystick,video,X]
+	media-libs/sdl-net
+	media-libs/sdl-sound"
+RDEPEND=${DEPEND}
+
+src_prepare() {
+	epatch "${FILESDIR}"/${P}-gcc46.patch
```
Comment (monsieurp) [inline games-emulation/dosbox/dosbox-0.74-r1.ebuild]: Stick those patches into the PATCHES array and then call `epatches "${PATCHES[@]}"`

**Comment by monsieurp** [comment]
@mgorny can you review those changes?

File: games-emulation/dosbox/dosbox-0.74-r1.ebuild (hunk 1 above)
Comment (austin987) [inline games-emulation/dosbox/dosbox-0.74-r1.ebuild]: The games eclass doesn't support eapi6:
austin@austin2:~/src/gentoo/games-emulation/dosbox$ ebuild dosbox-0.74-r1.ebuild clean install
Appending /home/austin/src/gentoo to PORTDIR_OVERLAY...
 * ERROR: games-emulation/dosbox-0.74-r1::gentoo failed (depend phase):
 *   base.eclass is banned in EAPI 6
 * 
 * Call stack:
 *               ebuild.sh, line 624:  Called source '/home/austin/src/gentoo/games-emulation/dosbox/dosbox-0.74-r1.ebuild'
 *   dosbox-0.74-r1.ebuild, line   6:  Called inherit 'eutils' 'games'
 *               ebuild.sh, line 320:  Called __qa_source '/home/austin/src/gentoo/eclass/games.eclass'
 *               ebuild.sh, line 120:  Called source '/home/austin/src/gentoo/eclass/games.eclass'
 *            games.eclass, line  32:  Called inherit 'base' 'multilib' 'toolchain-funcs' 'eutils' 'user'
 *               ebuild.sh, line 320:  Called __qa_source '/home/austin/src/gentoo/eclass/base.eclass'
 *               ebuild.sh, line 120:  Called source '/home/austin/src/gentoo/eclass/base.eclass'
 *             base.eclass, line  37:  Called die
 * The specific snippet of code:
 *   	6) die "${ECLASS}.eclass is banned in EAPI ${EAPI}";;
 * 
 * If you need support, post the output of `emerge --info '=games-emulation/dosbox-0.74-r1::gentoo'`,
 * the complete build log and the output of `emerge -pqv '=games-emulation/dosbox-0.74-r1::gentoo'`.
 * The ebuild environment file is located at '/var/tmp/portage/games-emulation/dosbox-0.74-r1/temp/environment'.
 * Working directory: '/usr/lib64/python3.4/site-packages'
 * S: '/var/tmp/portage/games-emulation/dosbox-0.74-r1/work/dosbox-0.74'

File: games-emulation/dosbox/dosbox-0.74-r1.ebuild (hunk 2 above)
Comment (austin987) [inline games-emulation/dosbox/dosbox-0.74-r1.ebuild]: Sure.

**Comment by austin987** [comment]
@monsieurp @mgorny updated

**Comment by monsieurp** [comment]
what was decided yesterday after talking to Mr_Bones? I eventually left and didn't follow through the conversation.

**Comment by austin987** [comment]
@monsieurp I never got a clear answer. AFAICT Mr_Bones doesn't want to revbump because that would make non-clang users rebuild for no real benefit. Alternatively, he doesn't want to commit to a stable ebuild, since the 9999 compiles. This of course leaves clang users with no stable option.

I was hoping someone else on games@ would review/commit. Otherwise I planned to ask comrel to mediate, since I've wasted more time than I care to to convince a maintainer to fix a bug rather than ignore it..

File: games-emulation/dosbox/dosbox-0.74-r1.ebuild [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	media-libs/sdl-sound"
+RDEPEND=${DEPEND}
+
+PATCHES=(
+	"${FILESDIR}"/${P}-clang.patch
+	"${FILESDIR}"/${P}-gcc46.patch
+)
+
+src_prepare() {
+	epatch "${PATCHES[@]}"
```
Comment (SoapGentoo) [inline games-emulation/dosbox/dosbox-0.74-r1.ebuild]: add `-p1` here, in order to futureproof all patches, such that one day these can be switched to `eapply`

**Comment by austin987** [comment]
Fixed by https://github.com/gentoo/gentoo/commit/143e036505e2b27b1aa5037d52a5f999391b4168

**Pull request closed without merging**